insights/specs/default.py

```diff
@@ -503,17 +503,33 @@ def httpd_cmd(broker):
     last_upload_globs = ["/etc/redhat-access-insights/.lastupload", "/etc/insights-client/.lastupload"]
     lastupload = glob_file(last_upload_globs)
 
-    @datasource()
-    def ld_library_path_of_pid(broker):
-        pids = [p for p in sorted(os_listdir('/proc/')) if p.isdigit()]
+    @datasource(HostContext)
+    def regular_users(broker):
+        """
+        Returns: The username of all regular users which could log in.
+                 But do NOT collect or store them.
+        """
+        ctx = broker[HostContext]
+        return ctx.shell_out("/bin/awk -F':' '!/nologin|false|sync|halt|shutdown/{print $1}' /etc/passwd")
+
+    @datasource(regular_users, HostContext)
+    def ld_library_path_of_user(broker):
+        """
+        Returns: The list of LD_LIBRARY_PATH of each regular user.
+                 Username is NOT collected or stored.
+        """
+        users = broker[DefaultSpecs.regular_users]
+        ctx = broker[HostContext]
         llds = []
-        for p in pids:
-            with open('/proc/{0}/environ'.format(p), 'r') as fp:
-                vars = fp.read()
-                lld = [v.split('=', 1)[-1] for v in vars.split('\x00') if v.startswith('LD_LIBRARY_PATH=')]
-                llds.append("{0} {1}".format(p, lld[0])) if lld else None
+        for u in users:
+            ret, vvs = ctx.shell_out("/bin/su -l {0} -c export".format(u), keep_rc=True)
```

Comment:
@xiangce should you use the `env` command instead of `export` to ensure that it will work regardless of the shell configuration?

Reply:
Done